common/lib/util/eventemitter.js

```diff
@@ -147,5 +147,30 @@ var EventEmitter = (function() {
 		}
 	};
 
+	/**
+	 * Private API
+	 *
+	 * Listen for a single occurrence of a state event and fire immediately if currentState matches targetState
+	 * @param targetState the name of the state event to listen to
+	 * @param currentState the name of the current state of this object
+	 * @param listener the listener to be called
+	 */
+	EventEmitter.prototype.onceOrIfState = function(targetState, currentState, listener /* ...listenerArgs */) {
+		var eventThis = {event:targetState},
+				listenerArgs = Array.prototype.slice.call(arguments, 3);
+
+		if((typeof(targetState) !== 'string') || (typeof(currentState) !== 'string')) {
+			throw("onceOrIf requires a valid event String argument");
+		} else if (typeof(listener) !== 'function') {
+			throw("onceOrIf requires a valid listener argument");
+		} else {
+			if(targetState === currentState) {
+				callListener(eventThis, listener, listenerArgs);
+			} else {
+				this.once(targetState, listener);
+			}
+		}
```

Comment:
You don't even need the first two `else`s because the first alternative never continues.

````
if() {
  throw
}

if() {
  throw
}

if() {
} else {
}
````